t/queue.t

```diff
@@ -1,391 +1,397 @@
-        HTFeed::Queue->new->enqueue(volume => testvolume, status => 'ready', priority => $priority);
-        is($priority,volume_in_feed_queue(testvolume, 'ready')->{priority});
-      };
-    };
-
-    describe "with an item already in the queue" => sub {
-      before each => sub { 
-        fake_bibdata(testvolume);
-        HTFeed::Queue->new->enqueue(volume => testvolume, status => 'ready');
-      };
-
-      describe "without the ignore flag" => sub {
+    my $testlog;
 
-        it "logs an error" => sub {
-          HTFeed::Queue->new->enqueue(volume => testvolume, status => 'ready');
-          ok($testlog->matches(qr(ERROR.*Duplicate)));
-        };
-
-        it "returns false" => sub {
-          ok(! HTFeed::Queue->new->enqueue(volume => testvolume, status => 'ready'));
-        };
-
-        it "doesn't add a message to the message queue" => sub {
-          my $receiver = HTFeed::Bunnies->new();
-          $receiver->reset_queue;
-
-          HTFeed::Queue->new->enqueue(volume => testvolume, status => 'ready', ignore => 1);
-          ok(!$receiver->next_job(RECV_WAIT));
-
-        }
-      };
-
-      describe "with the ignore flag" => sub {
-        it "returns false" => sub {
-          ok(! HTFeed::Queue->new->enqueue(volume => testvolume, status => 'ready', ignore => 1));
-        };
-
-        it "doesn't log an error" => sub {
-          HTFeed::Queue->new->enqueue(volume => testvolume, status => 'ready', ignore => 1);
-          ok(!$testlog->matches(qr(ERROR)));
-        };
-
-        it "doesn't add a message to the message queue" => sub {
-          my $receiver = HTFeed::Bunnies->new();
-          $receiver->reset_queue;
-
-          HTFeed::Queue->new->enqueue(volume => testvolume, status => 'ready', ignore => 1);
-          ok(!$receiver->next_job(RECV_WAIT));
-
-        }
-      };
+    before all => sub {
+        $testlog = HTFeed::Test::Logger->new();
     };
 
-    describe "without bib data" => sub {
-      describe "without the no_bibdata_ok flag" => sub {
-        it "logs a warning" => sub {
-          HTFeed::Queue->new->enqueue(volume=>testvolume, status=>'ready');
-
-          ok($testlog->matches(qr(WARN.*bib.*data)));
-        };
-
-        it "returns false" => sub {
-          ok(!HTFeed::Queue->new->enqueue(volume=>testvolume, status=>'ready'));
-        };
-        
-        it "doesn't add a message to the queue" => sub {
-          HTFeed::Queue->new->enqueue(volume=>testvolume, status=>'ready');
+    before each => sub {
+        HTFeed::Bunnies->new()->reset_queue;
+        get_dbh()->do("DELETE FROM feed_queue");
+        get_dbh()->do("DELETE FROM feed_zephir_items");
+        get_dbh()->do("DELETE FROM feed_queue_disallow");
+        $testlog->reset;
+    };
 
-          ok(!HTFeed::Bunnies->new->next_job(RECV_WAIT));
+    sub fake_bibdata {
+        my $volume = shift;
+        my $sql = join(
+            " ",
+            "REPLACE INTO feed_zephir_items",
```

Comment:
I wish there were a satisfying way to stick SQL multilines in source without jumping thru hoops. I hate all of them. Have you tried using a heredoc, and comparing side by side? It's what I have tentatively settled on with Ruby. There's a trick in the Perl Cookbook for cleaning up the indentation, using `s/^\s+//gm` although for SQL I don't think it's really needed except for debugging.

Reply:
(Note, CRMS still uses a lot of
```perl
my $sql = "SELECT something FROM blah" +
  " INNER JOIN spam" +
  " ON ...";
```
and getting rid of it is my life's ambition.